src/openrct2/management/finance.c

```diff
@@ -125,17 +125,19 @@ void finance_pay_research()
  */
 void finance_pay_interest()
 {
-    money32 current_loan = gBankLoan;
-    uint8 current_interest = gBankLoanInterestRate;
+	// This variable uses the 64-bit type as the computation below can involve multiplying very large numbers
+    // that will overflow money32 if the loan is greater than (1 << 31) / (5 * current_interest_rate)
+	money64 current_loan = gBankLoan;
+    uint8 current_interest_rate = gBankLoanInterestRate;
+	money32 interest_to_pay;
```

Comment:
only when
- current_loan == 0xFFFFFFFF
- gBankLoanInterestRate > 3276,8

then the result of `(current_loan * 5 * gBankLoanInterestRate) >> 14 ` would also yield a 64bit value.

Ultimately `finance_payment(interest_to_pay, RCT_EXPENDITURE_TYPE_INTEREST);` would crop the value back to money32 - without modifications.